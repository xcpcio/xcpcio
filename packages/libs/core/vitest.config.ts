import { defineConfig } from "vitest/config";
import { alias } from "../../../alias";

export default defineConfig({
  resolve: {
    alias,
  },
  test: {
    testTimeout: 30_000,
    name: "core",
    globals: true,
    include: ["test/**/*.{test,spec}.?(c|m)[jt]s?(x)"],
  },
});
