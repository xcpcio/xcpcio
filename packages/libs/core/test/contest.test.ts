import fs from "node:fs";
import { resolve } from "node:path";
import { createContest } from "@xcpcio/core";

import { describe, expect, it } from "vitest";

describe("contest", () => {
  it("2023_ccpc_final", () => {
    const data = fs.readFileSync(resolve(__dirname, "test-data/2023_ccpc_final/config.json"));
    const contestJSON = JSON.parse(data.toString());
    const contest = createContest(contestJSON);

    expect(contest.name).toMatchInlineSnapshot(`
      I18nText {
        "fallback": "第八届中国大学生程序设计竞赛总决赛（正式赛）",
        "texts": Map {},
      }
    `);
    expect(contest.startTime).toMatchInlineSnapshot(`"2023-05-14T01:10:00.000Z"`);
    expect(contest.endTime).toMatchInlineSnapshot("\"2023-05-14T06:10:00.000Z\"");
    expect(contest.freezeTime).toMatchInlineSnapshot("\"2023-05-14T05:10:00.000Z\"");
    expect(contest.penalty).toMatchInlineSnapshot("1200");
    expect(contest.organization).toMatchInlineSnapshot("\"School\"");
    expect(contest.problems).toMatchInlineSnapshot(`
      [
        Problem {
          "balloonColor": {
            "background_color": "rgba(189, 14, 14, 0.7)",
            "color": "#fff",
          },
          "id": "0",
          "label": "A",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "#951FD9",
            "color": "#fff",
          },
          "id": "1",
          "label": "B",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(255, 255, 255, 0.7)",
            "color": "#000",
          },
          "id": "2",
          "label": "C",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(38, 185, 60, 0.7)",
            "color": "#fff",
          },
          "id": "3",
          "label": "D",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(239, 217, 9, 0.7)",
            "color": "#000",
          },
          "id": "4",
          "label": "E",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(243, 88, 20, 0.7)",
            "color": "#fff",
          },
          "id": "5",
          "label": "F",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(12, 76, 138, 0.7)",
            "color": "#fff",
          },
          "id": "6",
          "label": "G",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(156, 155, 155, 0.7)",
            "color": "#000",
          },
          "id": "7",
          "label": "H",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(4, 154, 115, 0.7)",
            "color": "#fff",
          },
          "id": "8",
          "label": "I",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(159, 19, 236, 0.7)",
            "color": "#fff",
          },
          "id": "9",
          "label": "J",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(42, 197, 202, 0.7)",
            "color": "#000",
          },
          "id": "10",
          "label": "K",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(142, 56, 54, 0.7)",
            "color": "#fff",
          },
          "id": "11",
          "label": "L",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
        Problem {
          "balloonColor": {
            "background_color": "rgba(0, 0, 0, 0.7)",
            "color": "#fff",
          },
          "id": "12",
          "label": "M",
          "statistics": ProblemStatistics {
            "acceptedNum": 0,
            "attemptedNum": 0,
            "firstSolveSubmissions": [],
            "ignoreNum": 0,
            "lastSolveSubmissions": [],
            "pendingNum": 0,
            "rejectedNum": 0,
            "se": 0,
            "submittedNum": 0,
          },
        },
      ]
    `);

    expect(contest.statusTimeDisplay).toMatchInlineSnapshot(`
      {
        "correct": true,
        "incorrect": true,
        "pending": true,
      }
    `);
    expect(contest.medal).toMatchInlineSnapshot(`
      {
        "official": {
          "bronze": 4,
          "gold": 4,
          "silver": 4,
        },
      }
    `);

    expect(contest.getContestDuration()).toMatchInlineSnapshot("\"05:00:00\"");
    expect(contest.getContestElapsedTime()).toMatchInlineSnapshot("\"05:00:00\"");
    expect(contest.getContestRemainingTime()).toMatchInlineSnapshot("\"00:00:00\"");
    expect(contest.getContestPendingTime()).toMatchInlineSnapshot("\"00:00:00\"");
    expect(contest.getContestState()).toMatchInlineSnapshot("\"FINISHED\"");
    expect(contest.getContestProgressRatio()).toMatchInlineSnapshot("100");

    expect(JSON.stringify(contest)).toMatchSnapshot();
  });
});
