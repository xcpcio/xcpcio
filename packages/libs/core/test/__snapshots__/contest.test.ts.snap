// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`contest > 2023_ccpc_final 16`] = `"{"id":"","name":{"texts":{},"fallback":"第八届中国大学生程序设计竞赛总决赛（正式赛）"},"startTime":"2023-05-14T01:10:00.000Z","endTime":"2023-05-14T06:10:00.000Z","freezeTime":"2023-05-14T05:10:00.000Z","totalDurationTimestamp":18000,"freezeDurationTimestamp":3600,"unFreezeDurationTimestamp":14400,"penalty":1200,"problems":[{"id":"0","label":"A","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(189, 14, 14, 0.7)"}},{"id":"1","label":"B","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"#951FD9"}},{"id":"2","label":"C","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#000","background_color":"rgba(255, 255, 255, 0.7)"}},{"id":"3","label":"D","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(38, 185, 60, 0.7)"}},{"id":"4","label":"E","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#000","background_color":"rgba(239, 217, 9, 0.7)"}},{"id":"5","label":"F","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(243, 88, 20, 0.7)"}},{"id":"6","label":"G","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(12, 76, 138, 0.7)"}},{"id":"7","label":"H","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#000","background_color":"rgba(156, 155, 155, 0.7)"}},{"id":"8","label":"I","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(4, 154, 115, 0.7)"}},{"id":"9","label":"J","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(159, 19, 236, 0.7)"}},{"id":"10","label":"K","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#000","background_color":"rgba(42, 197, 202, 0.7)"}},{"id":"11","label":"L","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(142, 56, 54, 0.7)"}},{"id":"12","label":"M","statistics":{"acceptedNum":0,"rejectedNum":0,"pendingNum":0,"submittedNum":0,"attemptedNum":0,"ignoreNum":0,"se":0,"firstSolveSubmissions":[],"lastSolveSubmissions":[]},"balloonColor":{"color":"#fff","background_color":"rgba(0, 0, 0, 0.7)"}}],"problemsMap":{},"statusTimeDisplay":{"correct":true,"incorrect":true,"pending":true},"group":{},"tag":{},"options":{"calculationOfPenalty":"in_minutes","submissionTimestampUnit":"second","submissionHasTimeField":false,"submissionHasLanguageField":false,"submissionEnableActionField":false,"submissionHasReactionField":false},"organization":"School","medal":{"official":{"gold":4,"silver":4,"bronze":4}},"awards":{},"logo":{"base64":"\\niVBORw0KGgoAAAANSUhEUgAAAL0AAAC9CAMAAADRCYwCAAAB41BMVEVHcEwau034r2WEe974tmmF5WgnwlPgv5Bz5WT7vHPxe1GRheFscds1Tc8sxlMlxU0/0FZQ3lf9tzxJ3VnyXTXzXjdT3138tTvyWDT5uHErxVQ/UdHxTyo8z1hGzlr9tjxhZ9nwQyHyeFFna9r5rWMauE9Y1mD0e1NPXtMXt0tQ22Atz079uUT9sS1+ed3zXDlL1FgtyFL9tznzcUrza0X9sy5MXdROX9TvOhtLVtbzYT38u184yFo1Tc5UYNb9vlNcZdY1z1Ni1GP9sCn9tzspSMpGWNJh22Ks6X31jGT0b0v5wolV0Gaiwidycdv9tzn0dU79vlE5T9AYt01q32X8v1BeZ9fzdFB6326M5XBDVdJaZNb8v1MPrEyBfNzwSSX8vFmz54R23W/2lG79tDSYi+D3yGXzfVj0e1SfUZxtctn52a2hleH7q0OT3on727bwy22r1DFSuKboztr7y3UwmKSroNv+sSYoR8rvMxYRsksoRcsTtEkRskn+sSQPsEsMrUzvNRclR8kPsU7xMRT/tCfxNhgVtUYJqU30NRcZuVIoSs71PyD/ty4yTtD2TSv/uTw9VdLvuR9wvi6hOXPKN0lfO6VBQrtMuzflMyvMvyLP3agpgLEgsHaZylH/0IGVjSRsRSpHAAAAd3RSTlMA2icvHg7+BxsTNhAf59q3szWEQZQjJFXiZOoz7ZfarYv4U0dF8K6X8LBRScBkbPl8yuhpqPrf/uVox3n59XY1zWLS19n3xGwqjvM2/v1Z9H6Y1D9Z/aHkjEibt9L2p8btV67cl5Czyrj26EjI5ub45K/0t49u/SxprvcAABVaSURBVHja7FtrT9taFiWEJBASEtSAAgTEo7xaiYCCeN2CKBJQYFQhWkFbRDvDVQW9d3Tnw3xJHFs2tuXEDyVCglbVSKP5qbP38bET3okdJSB1owKF5GR5n7XXXvvENDX9il/xK2oYbS0tXVa0tLQ9JeiB4PDJQTy+YEYifvB7KtgWeBLQW7pSSwnDUFRFVWX4UBRFNYzE0niw+dFjDwJ0RZUzEGdm4HeZDFxM4iD1/FFvQDAWN2RZMVGflX+BkGUjGXv+eLHPAnYbbQm2HRnZiJ8+Tvxtu4ky7HcE4E9uPT4FCvQsDakPYSf4leJs12NL/GzxOnYsXFnOlBcw5b/yJdXyuBhvyNcYIqtG0QzDUOUrlMqoxdNHxJ5g3MiUyhSQK8XkQSw2voUxHovNbhdBikroz+Ti7mPRzkAqXsaajHyW2B4f7tkp51XP8O5B4kwtiag89EjIH9hNlsCjJo53Nd9MbKC5a7dcT2Xj9FFwfzgp27QBPZxtvpMTzacJxeaPXBwPPAbOy6VqXFgavg9SW3CpKNtXOtR47gfjagl8MvXQw5vHF2yaKcXhRuv8kg0mYyxVYgKCXxQbfjzY4CZl63zGiFVWhi1fbO0xlhpaubtFC7w6FKuUxcO2wMpGIys3mKA1qMiVgyfwrUppIHcCMcXK/EKqmiSCyNInGrGGoU8NURDyUJXqsWs9U1kYbhhvLAIosSrpG5g1MpQ7S41hvo3gTN2umr3NJvXRr6UaV7KEN4nqdz+wVbSSP9sQs2xvvrHr5NzkgNJOTjaC+c1f6MurzlpOykq+0Qj0u1bqi1vOrv7AIl4DqBNwmfqS3sqJ+s8pw9QjZIa2nFZ90lphuGE1q8adGq22WYpePam35P+5TV9acc7aFB0U69+weuxtd26zUkWK/ku90acsxUk+d52BBqCnA1LmYMfxGi1LKnVqdZ5RLLF21metiJkbmDG26os+uF0LtbPRj9cZfbIWnaZh6Km/lOPNTxi96gr9OBWuuqO3PMp2LXJf3Hqy6DMZeeH5E8593buVhV5xV7X4XnQmU3/0tdGcxMLCwplad5dWG71vw1swtk5O6n0cWJteaw5pEHU2aTsW+npLdU2i7VQx34SVt3eeIHzL358la6LVvlZf5/HxWBRj7PhDt6+mxwzAzVAfDcJTe7YyanCQ53u/NuDx+DVNYzUMv6djbdNXG+TtoVDf4cZGeJ3GxsbhVKhn23z3OKMeuBstfN3d/R2aJDGSwPECBCcIPCtJR57ou263V9A+MrI/EQ6ndT2b1dMYuq4X1sMT3y7PVOUMPhKuqNM53Tt6JPA8x3A8w7IcD5/hO/giCUe9/S8/utGjt/vzBVFMp7Np/AcfWfO7tCj++H7xU1WBOqfOl9/s72XyvMQCbIbjGDaXY3Msy3A5lmPhu3xe+9u0U/jtU4dhHaBDzgloK8z/iHoW8MuK7PjNm9Z+vyZwDMPnWMaEz3HkE8vgXsBVAItGI5uO8IcOz3XMu5n4LPlnXwRugyjqiN/hmzeB7ogm8AzSBdPOwQ4AZuA8/EQi/OFwA3jJH+2seu3Q4J5uQaef6AZYP8Gvevr7ZSbhpN92jnkEnmNzABw/oGJ5IBDDMJLEcwIvIXQGNoZjBanjWWt1eV+ZWNfTNtisxfh02QWY/NfTFz8dnMN2zmg8wIU8Y/qFvMBommdyMhKZnJwc1VhQH7gSuDb4PVzX6NrH6khj5dsGK9qRzdq5J/i/X1YLP/DSg3nNcTxSnBc0f+/M2gcrwaD+HX6NFxjEz2MNCNrAs4oXH9nQTZ2hOAG5Xiicz2OcnxegGiinyIYg/Orul2jt1/ICpJVFenN5tjd6/A9fmbYHfG8+jPXCY5A4LOoPL0UqJE9gao4wvhTr86uLi4tTZiwurqxOLKfFsvyLP779Vg34iMazKCkCn8vlpVf9t9Li4/SAlodGQKoB+BN5WZFODs6nRUsaIbfre6sj3tCVh3i9fStzsAVZa4MAfuWV2z0pAeNB2lEdGf90510d9c3mANEgxsx+byXwp85NXhDGZAsTg+3tN7enKdA+NQeCaimpmP5W6U26m708ch61nZdGo8/ueVrA99ojMahKTI6pCP5IOGvro1iYGwzd+Ujv572Cbsm/mP32WyXwA5t+CXNOWqngef3gRiHLiDSBd5jpfqi9hs1ORBxNeN/7cE+gXUD8cVFB6QY++EELcwy2IkGLdlfAs6gmMKicwB8hem/pBr7OW50UEr8x8uDafdjSCP2z4o/LB2/zQ/DYXdGOSZ7NNxWVyfSoAI/PoURp/ffq/JxNhfT6PaQpo89cQcyatat//xl/4BbjZ68IeAbIIL16X2lzeOERSN+VAP59VPtk6UhaX16pbG3vasHuWxdq8SR4D/7WGWIAQEI4oaPy9tMEO8aiaWYloeNOsgVWlm0bFl6pdGkC32xrPy5lNXEyvHPHBbROgoDkOLRgVYFvenOs5dE04wRwZ/LfnltNSt/rq3xt7wZtbtB0fyqKOrQ9m+ppuxW8hG0fWo9UFXh46pjGg2aiIxp9fxeFiT+ANOrhvmrW9lrVIqYvVLwdVikml2Kx2DhGbJzWQncEbBmHE5QgvHpR7fzYLxFnwXFS9FZla98viJQ46yvVrT2yTJ0/6I5KbwBXDXoLuHHQTDnPY9thkfPVgscGDc/OoZMevfXJ3gnqyUR9pdq1BwvUtukXsmLe6kpuvicR7yLeBqcn0jeFKmljxmspj24HWDdzi7MIHFpSqW+0VzvKtH8Sqd+Ewr1yGz4e1HYheInJIXMk3hn4Jl8EpxgcVkY7b3MI1iQSHql+7VCYWiNI/rW/1FATXdAwsV6BtZwg9ToCD/5IwyYN3NGiN3+5X6CVd77iYOn2Q536HZCda+gXuroHJCLYMI044TzVnQ4BdQeKd+ZGj/aGral11etk7RFQW6JY6Uv56p+ZKAtBsCrAenhhsAcOMw/0nAZjDVMvK/g3b6be7LLi+Vdnpyf7lmp+SxpnMvlTGfOP3OShNXhZ6FLYYZ2DB+bPSAweoAja9HXe7tEX11dDztZeOTdTL57/51+/byeGhgwaxf8RVwm2UnLMeTOiEp6cwEr91yR/kLZZp6lHwTVnRbHwzz93doLBnt/N+Pt/BQEPmSBvUu97N+CbXhzhoQmXE159vCHYZAxMz4Wcrv1JN4+r9FVTcNtIvFkDzrM4i7CSx1XmoW495JiEzfufXbMqItX6FcdrvyWVgx6prF20RiU05zAI8oJb8Kg65PiH066OiG/nqT8Lh5yjnzepU47eF9WwwYM5Z11ynqAfyONYxrLCu1vzlp5rd7x2aIMeMeyFSt5KE3LkPEMS/H+9afL5fG4OtFvX8uBzYCfz/VfMwqJl6w+do0fiE/otfy3ztdjeuRyMRTm/Z2ZmZmDaTfJfI3qQfGGgfL4N/UFN1vqIi7U/FahsfbbBm4dhPOh8L7hzXhDyk27QvztizGPPK+i9f5gnA+Jyn4u1Bwtmvyt8LmUez8FA6Ds6wehAsxVYd+hxuoI6yne0XtNqAr+G6AP9GlgDCd9cEHpfNEVAezgYrF2jx7K9hj5MO82E1xV6sYQeaQOdUeIl0x5EMPdg712ih9TDKlfR94Vpp3WPHgLRtx5r6CoZFHqi8xHyfohb5hxJ+IYQcwt69Ciu0ROj87nJN+bn8OwDPDHV+YhEJjt3uZ/OYz54Jj9wS+6zNcl99vwrcp5ULBCd+vmIRFymK963rOUZJCN79USwdszB4jn/N6ENx+XQz9MxLoKZd1m10Gt5Dscr4YpTqA1zPuGBiiiK83/9v72r/0kjW8OoFCQoFZpo6I3FxGK6ikZJ0DSo+MFtTRQNXdmsvbpxN64/bJvtdu/HmIFhTphMZjJkyKi/mLa7/Vfv+55zBvEbGC/uJPckKv3I8MyZ933e5/044iO07Yp63s5BwWtpS7BrxwH6EBFRrEqlCc8lzqGF1xdOGHNx9s1CrVY9+oKqEoU46PnwuTYnVKH4vneiMWVIjTE9eXJJm1Nx6yxa+YPB0dGf//O7ImHBDJQs8rz9b2O0N3tNXtRKXr4HngO6Sb+o7yHWsmqSI/QM5q8+dFis99VtnhZSFYm2qkoO0ENuhbQlys/WLialts5ZdIr+W3i6CiTPsEWN4D2DXbSECW7btsp81KVAyIPs/vLze8lr2EdLQafgJQmLToA0FL5cz8Ai8DUFgabZHlwHxCr4zvSVvIildW/6HIEfxyaZRDsjT8cvsR3BNiEY/li7W/+shHxZ0qTQJd4a5TKtWBtxAv5JHPYX+5Pi1TTwNRYzsEf7bK1Nn43TJm+JXKml+TGzwGZbdbbPidkgPhH1WNeVWuMTHwGqxpS6PdPpHRBlFPcaMb6/IU4CZ045AU+xS0QJXS2UDoLpoE2JykBb81qvsftTQtH97ErjcGqOc2Ztyd+2zdPugoR6/roEfBpyXKyBy75wG1cfnEG+0cB0rm69J/iqSkvYxaP+NpPDibiCdUb5gs0HoukIXel0+nBDx2RF0shw6xsEuo8oOMyjkevK94vHtI5ZrNTetrP53p+6aC8TqFKnNh/wPj+MxRJblsmWZVk5DWux8H+M1tXCNtqNiAotfl27Hzef6cyFl2241E8+BKZhGjj/xBNYf57IJk1TVct8CfB1cqYj6aBltWo7409xWAfb1OTxtV4z9XdeAq/Otiw0IRmh24rdhfnP/4weFlKqjVwQBPt7TkPbBeV558DBFY9XMKGFO9d915f/g0tVXtSptlrUYXUbSYEwq89//vDVMgWKFr6Ec/CCoJ7qSNiiIhsDvS2RJU6aYhwhysr10cI/tVDh0za171qCj3MQbIaS6Bufv1p8x9mqmw6+/qQRjDhAHcZY8/DXxrB5gaH6svy4wPlsBhA89/iHYCvgFUzYMIfVN34rCDb0+t7z1/CHco4Q2rxBwdAs6/+4YtBWKRgOid/cNhp9UWXjrkeVuear+GGwSWRKEcCfFUzbSuh2c3e1HRf+4lSnKZZMZGO4OfhrK4Yu0qqWfPuICJvNocTTvxlsNooYBKc+wG700xOKUKhvehl8lzMm9WJBQN5RsH+lNDknN0iLESgSgBZu9xba9WRrrinjCYx/Q4dXsVTDwQsCvwFBNa1E7HxlLVNF09epYABMyuPxu7Z/bfsxgeit4XQRufN239bskbTKcROuGx7wyQrNNaVG8GzXsxBgGw8EZdKTB1sMPs6rYx7gG76dOcMrPkKweAaeLt39rHBegg/oHNX+0XcXUYYMnY0+K4QAeNxztu2CmZ3MXKU1fyadVHOajnOkOClKjNDwzZD8yyGR4IAyzlfAzt8dI6Ze1IdfK7U3L2/Zfv/4jAGPn45VinTn60tNFfI3Dcd2H27lwGclmWYCsmLM3GA+j6YfG9gpxQKIpjXpJaNUrrE2QnVhabPvetXT+3omrhAFSF6jNp9TbV4E7Nn0+s0GHcjkcxKRZUJHYSEJjs8sD65dhNa7Nrg8swcPFAcBcfJRNpochcVZIT7Di/hnFzd7rvrSMjINCmLs6ki6TMHzgJqMeG/Xef7ujz5d1tAVMZuRyV58Y2V5e/tHtrbHl1fm43sE58NlVHXI88PNBreepVqlPuV9VK3NLY2MjDbgCU+vzJd0GZkAK6LAlCKC5/HUzGaaUKQfz3TYeyJqOOmqgMuLhm9+g655nw+eJahhUdEkepqAKHePbTak6L/wciobWq9Ua8cv3r60179nfKLOSqwaBilZ9/1p1t3VTGSaeYtHH091XaJkKFPfgVCn45EZHX9QSSMpEBOwhCIroZYkac9if/28ADvjUMGjMnRVv2BdAuvBVLGKEugyi4dXQbASTR6a9B4i9dA5rxK7DgKmE74oCUo4/yor+BZ6fLhVPT06W600HntgAbgC91H8ouFmARWg4UjE2Phc4BYvlM1E0yc+vYefTgnVR4z6JaoG6Mirxu4Fh3dlIj4Ot54G9ywdV4v2AQ17Up3WTL4Y9GQLvgshytNfP3ytBykz38IJtUDezJ0R9B8ZmRMfAzusUcJSOT05QxQjtNxWCh8ceVWjM8bFYrHxwE+xZgB2BQhbl42nw4PvPlhlzpVmoqUTEIGEepI7Q3vHG0BWL9ERX3isogLupJfiM2ODnjZXcPFVDU9pnENnY39/4GQcXNz4ZngbMOyqnCtbMBtuPAnIFnNnGlxLEfm8oIhDvrAzOoGtmXbUWe/bfLtQrJ9Z4icFAH1J1kWIMUjB/nSKU72abfmU81AKiOrkU+70TEOu0RnnwKu9+PzA8o7jU7fBv/0wd0yH7KtFel4GnOEPIvpC02EWPqIFroZVK9ry1QMxk0bok5NPfyLP+wwDvs1vjH0bDvd67mMFR0dHfpnt7+9fOD4+rsHP2d/HJibq5pg2baPPt3HxtMVzGXN3Z2eCr52dezrdySN7H67N9+/f/9wDL3obD+ccqJwuD9o5EOtPmIyv1MT/+nfQXHf++3mKJVPqanvv3p3lPm+lPZ1feSYty+pum2eRIyZDb0Y6D96bZIZTTkXbvcKqytRdovOfSwCUxwhnv92j8QE0HdR41lDHt35X5cqy7efuj/GsQOg4+swWqziVC9H2rwHRjj6/jqOPphjXqzG/gx3giU2kw7/QIvCbyQKtOeRxhJ56/n6H3Tawa7LCU8rBqf71ff78kp1Gz0ONE8PxdO8yqdF59KuM7VVHkWZSZWnlg6CHfbsP9GXhAdCXnbE9Q89o92Esp+xQo0zy5OYh0GNbwTn6B7J7+tDvZe87z/fMchzbPYMfCzwE5zj2WlNFpVPutM5hdi84ZExv5OBgv1CwHgQ90sWB0yutR6PRTn8ABCoFWhNI/nU/L+wOlYac2fm86J4UMsut3Ije/68U7YULaiLgQvg0M0TWsaIuRN+9r/LUIu93IfxoihfytjIuRO9N8p6POulGv83b3fGs151+y3Y/FXHj5h+onPK3hlzpt3a/MOFC26H9A9bij7kQfnSLZxeCdejCiBupj+W4MeKC7di06Ual/DyrnsN/5D74q7yeKQjJQ/e57qRpz2Cqlvv0mj9m8pALrnvgut33Us+1G8fpgCvhs5Kemkqk3fXJ9h4vNR4+/q2mkrtDATc5QCBvqQ1zvGYqm89kMutucQJvfrUBPqTqQmorub87yVb3Xx5/JmueH3gAF1BVe1TQYbGwQ1l6JKXWx8DtF3gPqhvQe/zRZEpthG97srrrht/x7++OZBsMhp/fgB+Fd+6Iu95ILMUbafZsMrwsuEZ+BoZiFjs9c25FlovEsz+d3y808D/IH3dJ/3fRoQvLjXXa/6+b138BVEE6mOU/ueYAAAAASUVORK5CYII=\\n"}}"`;
